pandas/core/generic.py

```diff
@@ -1787,6 +1787,10 @@ def _slice(self, slobj, axis=0, kind=None):
         return result
 
     def _set_item(self, key, value):
+        if callable(getattr(self, key, None)):
+            warnings.warn("Pandas doesn't allow attribute-like access to "
+                          "columns whose names collide with methods",
```

Comment:
i just want to see a more verbose message that this is not recommended
we could also raise but that's a bit harsh